### PR TITLE
Refactor tests to use a Prisma client deep mock

### PR DESCRIPTION
**Pull request by ggwadera**
- Updated `prisma` to 2.28.0
- Added `jest-mock-extended` to create a deep mock the Prisma client, this makes it easier to mock the entire client when testing, and also adds types that help when coding the tests
- Updated the tests to use this deep mock
- `prisma/index.ts` file was changed to only have a default export, this was required because `jest.mock()` mocks the default export of a file
- Also did some little refactoring here and there
  - Removed some try/catch blocks that did nothing other than just rethrowing the error
  - Moved the admin checks to a single function to reduce duplicated code and tests

Used this doc from Prisma as reference: https://www.prisma.io/docs/guides/testing/unit-testing

### CODE REVIEW

**Review by tomrule007 — CHANGES_REQUESTED** [review]
I like the changes! I believe either jest or next are leaving zombie prisma clients around so I'm guess this will help with that problem as well. 

Also nice having the one `checkIsAdmin` function with one test instead of the same code getting tested all over the place but I was wondering if `isAdminOrThrow` might be a clearier name? to bad typescript doesnt have a throw type.


I also saw some missed `= jest.fn()` assignments that I believe can be removed since the whole thing is mocked now.

One last question I have is it possible we could move the utility to the `__mocks__` folder and just get auto mocking? I'm guessing that it probably isn't recommend since you still have to manually set the responses to each command.

File: graphql/queryResolvers/session.ts [hunk 1]
```diff
@@ -36,12 +34,10 @@ export const session = async (_parent: void, _args: void, context: Context) => {
     })
   ])
 
-  const submissions = _.get(session, 'submissions', [])
-  const userLessons = _.get(session, 'lessonStatus', [])
+  const { submissions, lessonStatus: userLessons } = session
```
[the remark this answers is not in the document]
Comment (ggwadera) [reply graphql/queryResolvers/session.ts:37]: Yeah, Prisma will return an empty array already if the table join results in no corresponding values.

File: graphql/queryResolvers/alerts.test.js [hunk 2]
```diff
@@ -2,17 +2,17 @@
...
-    expect(alerts()).toEqual([])
+  test('should return empty array if no alerts', () => {
+    prismaMock.alert.findMany.mockResolvedValue([])
+    return expect(alerts()).resolves.toEqual([])
   })
 
-  test('should return list of alerts', async () => {
-    prisma.alert.findMany = jest.fn().mockReturnValue([
+  test('should return list of alerts', () => {
+    prismaMock.alert.findMany = jest.fn().mockResolvedValue([
```
[the remark this answers is not in the document]
Comment (ggwadera) [reply graphql/queryResolvers/alerts.test.js]: Thanks, missed some of those haha.

Did a search now and should not have anymore of those.

File: helpers/isAdmin.ts [hunk 3]
```diff
@@ -1,5 +1,9 @@
 import { LoggedRequest } from '../@types/helpers'
 
 export const isAdmin = (req: LoggedRequest): boolean => {
-  return (req.user && req.user.isAdmin) || false
+  return req.user?.isAdmin ?? false
+}
+
+export const checkIsAdmin = (req: LoggedRequest): void => {
```
[the remark this answers is not in the document]
Comment (ggwadera) [reply helpers/isAdmin.ts]: Yeah I agree, was kinda unsure on how to name this one.

**Comment by ggwadera** [comment]
> One last question I have is it possible we could move the utility to the `__mocks__` folder and just get auto mocking? I'm guessing that it probably isn't recommend since you still have to manually set the responses to each command.

Hmm I don't know about that. Since we need to mock the client from the `prisma/index.ts` file, this causes the import path to be relative to each file and I dunno how jest works with that when looking for a mock.

**Review by tomrule007 — APPROVED** [review]

**Pull request merged**